### PR TITLE
Incorporate key material text suggestions from Laurence

### LINKED ISSUE #7: key material for attestation
In the IEFT 117 meeting, Laurence suggested not using the word "identity" but rather "key material for verifying evidence" or "attestation keys".  He suggested there might be like 3 pages of stuff to talk about here.   I asked if there were volunteers to contribute such text and the recording shows that two people raised their hands (Thomas Hardjono was one, and I think the other might have been Henk).

### PR DESCRIPTION
**Pull request by dthaler**
This PR incorporates suggestions from @laurencelundblade 

Fixes #7

### CODE REVIEW

File: draft-ietf-rats-endorsements.md
Comment (thomas-fossati) [inline draft-ietf-rats-endorsements.md]:
```suggestion
with a private key that the Attesting Environment possesses and the Verifier will verify
```

File: draft-ietf-rats-endorsements.md
Comment (thomas-fossati) [inline draft-ietf-rats-endorsements.md]: one _"cryptographically"_ can probably be removed without loss of information, e.g.:

```suggestion
cryptographically verify that the Evidence
has been generated by the Attester that has the key material associated
```

File: draft-ietf-rats-endorsements.md
Comment (thomas-fossati) [inline draft-ietf-rats-endorsements.md]: extra comma (not in the scope of this PR, but since we in the area)

```suggestion
This document treats identity claims as with any other claims but allows
```

File: draft-ietf-rats-endorsements.md
Comment (thomas-fossati) [inline draft-ietf-rats-endorsements.md]: slight streamlining

```suggestion
A Verifier receives Evidence from an Attester that must be considered untrusted
until verified through cryptography. Typically,
```

File: draft-ietf-rats-endorsements.md
Comment (thomas-fossati) [inline draft-ietf-rats-endorsements.md]: The sentence _"[t]he Verifier must have a verification key for each Attester."_ may give the impression that a Verifier must be provisioned with the key that verifies the Attester's signature.  In reality, the requirement is less stringent: it is sufficient that the Verifier is given enough information to trust the verification key (e.g., via one or more CA certs) rather than to _"have"_ such verification key.  I.e., the actual key could be in-band with Evidence as long as it chains up to a trusted anchor.

File: draft-ietf-rats-endorsements.md
Comment (thomas-fossati) [inline draft-ietf-rats-endorsements.md]: What do we mean by _"more sophisticated"_?

**Review by thomas-fossati — COMMENTED** [review]
good stuff, thanks.  I have left a few comments and suggestions inline.

File: draft-ietf-rats-endorsements.md
Comment (laurencelundblade) [reply draft-ietf-rats-endorsements.md]: Yes. Very much agreed.

Would be good to add text explaining this -- that an endorsement might contain a root certificate or other key material that is needed to establish trust in the attester 

(and it would help get closer to my prediction that 3 pages were needed :-) )

File: draft-ietf-rats-endorsements.md
Comment (laurencelundblade) [reply draft-ietf-rats-endorsements.md]: Some new DAA crypto protocol.

**Comment by laurencelundblade** [comment]
Dave, thanks for turning my crude first pass into a PR.

File: draft-ietf-rats-endorsements.md
Comment (thomas-fossati) [reply draft-ietf-rats-endorsements.md]: awesome, should we add it as an example?

File: draft-ietf-rats-endorsements.md
Comment (henkbirkholz) [inline draft-ietf-rats-endorsements.md]:
```suggestion
the bottom-most (a leaf) Attesting Environment in an Attester will sign claims about one or more Target Environments
```

File: draft-ietf-rats-endorsements.md
Comment (henkbirkholz) [inline draft-ietf-rats-endorsements.md]:
```suggestion
the resulting Evidence with a public key it possesses, called a verification key below. While this is typical,
```

File: draft-ietf-rats-endorsements.md
Comment (henkbirkholz) [inline draft-ietf-rats-endorsements.md:211]: This sentence needs healing.

File: draft-ietf-rats-endorsements.md
Comment (thomas-fossati) [inline draft-ietf-rats-endorsements.md]: Sorry, I missed the change of title.
I think this "endorsing identity" is a valid and useful conceptualisation.
What we are talking about here is precisely the endorsements that allow a verifier to know that a certain cryptographic key "speaks for" a known and genuine attester, i.e., the mechanism through which its identity is anchored.

File: draft-ietf-rats-endorsements.md
Comment (thomas-fossati) [reply draft-ietf-rats-endorsements.md]: why leaf?

File: draft-ietf-rats-endorsements.md
Comment (dthaler) [reply draft-ietf-rats-endorsements.md]: I think the changed title is better since "Endorsing Identity" doesn't seem right when DAA is used but the new title and the text in the section still applies in that case.

File: draft-ietf-rats-endorsements.md
Comment (dthaler) [reply draft-ietf-rats-endorsements.md]: Yeah I don't get that either

File: draft-ietf-rats-endorsements.md
Comment (dthaler) [reply draft-ietf-rats-endorsements.md]: I was thinking an example of "more sophistocated" would be a certificate.

File: draft-ietf-rats-endorsements.md
Comment (dthaler) [reply draft-ietf-rats-endorsements.md]:
```suggestion
the bottom-most Attesting Environment in an Attester will sign claims about one or more Target Environments
```

File: draft-ietf-rats-endorsements.md
Comment (thomas-fossati) [reply draft-ietf-rats-endorsements.md]: IIUC, the DAA Issuer endorses a _group_ identity, but an identity still.

File: draft-ietf-rats-endorsements.md
Comment (thomas-fossati) [inline draft-ietf-rats-endorsements.md]: I'll give it a try:

```suggestion
of the Attester. The verification key could be, typically, a symmetric key, a raw public key, or a certified public key.
```

**Comment by thomas-fossati** [comment]
@dthaler @henkbirkholz @laurencelundblade:

The draft is going to expire soon (2024-06-03).

There are a few remaining review comments that need to be addressed before we can wrap this up.

Can you please try and have a look to see what's missing / what needs changing in your view?

It'd be great if we could reach a consensus in the next few days as this is the only PR we need to merge before we publish an update and ask WGLC on the document.

File: draft-ietf-rats-endorsements.md
Comment (dthaler) [reply draft-ietf-rats-endorsements.md]: The rest of the WG pushed back against the concept of "identity" per se being the focus.  Is there a technical objection to the change in section title?

File: draft-ietf-rats-endorsements.md
Comment (dthaler) [reply draft-ietf-rats-endorsements.md:211]: Can you elaborate on what the problem is?

File: draft-ietf-rats-endorsements.md
Comment (thomas-fossati) [reply draft-ietf-rats-endorsements.md]: If you are referring to the conversation starting at https://mailarchive.ietf.org/arch/msg/rats/hEnwNxvLHcLV7CQGmwQvYwztGqI/ 
there was some initial pushback from Henk and Laurence (Ned's points were more neutral), to which I replied to clarify my point further.
The conversation stopped there. So, from my perspective, it's still in limbo.

@laurencelundblade, @henkbirkholz can you please look at that thread again and see if my reply clarifies your initial confusion?

File: draft-ietf-rats-endorsements.md
Comment (dthaler) [reply draft-ietf-rats-endorsements.md]: Updated text.  Is the new text better?

File: draft-ietf-rats-endorsements.md
Comment (thomas-fossati) [inline draft-ietf-rats-endorsements.md]: hmm, providing the verification key material in the form of a certificate doesn't seem to solve the scalabilty problem, it's still 1:1 with the attesters' population.

If you have the right trust anchor configured, the endorsement consists of the intermediates' certs that allow completing the path to the Attester's cert, i.e.:

```
TA -> iCA1 -> ... -> iCAn -> attesters' certs
      '-----------------'
          endorsement
```
Note that this kind of endorsement could be inlined with Evidence, e.g., in EAT (or other CWT-/JWT-based formats) using an `x5chain` header parameter.

Comment (dthaler) [reply draft-ietf-rats-endorsements.md]: Scalability is solved by not storing the attester's certificate.  It still comes in an endorsement, but whether that is resolved on demand or passed along with the evidence depends on the protocol.

Comment (thomas-fossati) [reply draft-ietf-rats-endorsements.md]: > Scalability is solved by not storing the attester's certificate.

OK, then I don't understand what _"Attester's verification key material"_ means in the following:

> _"[e]ndorsement from an Endorser includes the Attester's verification key material in the form of a certificate that chains up to that trusted root."_

Does that exclude the leaf cert?

Comment (dthaler) [reply draft-ietf-rats-endorsements.md]: A certificate that includes the Attester's public key, signed by the Endorser.  I'd call that the leaf cert.  I'm saying the leaf cert need not be stored by the Verifier.  Thus the Verifier's persistent state need not be O(Attesters).

Comment (thomas-fossati) [reply draft-ietf-rats-endorsements.md]: > I'm saying the leaf cert need not be stored by the Verifier. Thus the Verifier's persistent state need not be O(Attesters).

That's the key point, but lines 217-218 are not making that entirely transparent. I suggest we explicitly state it.

Comment (dthaler) [reply draft-ietf-rats-endorsements.md]: Ok, added sentence, please check.

Comment (thomas-fossati) [reply draft-ietf-rats-endorsements.md]: Thanks very much.

File: draft-ietf-rats-endorsements.md
Comment (henkbirkholz) [inline draft-ietf-rats-endorsements.md:207]:
```suggestion
A Verifier receives Evidence from an Attester that must be considered untrusted
until its roots of trust endorsements and their corresponding identity documents are verified through cryptography. Typically,
```

Comment (dthaler) [reply draft-ietf-rats-endorsements.md:207]: I disagree with this wording as phrased because it incorrectly implies that endorsements are mandatory.  They're not mandatory (at the expense of scalability), as other text already explains.

File: draft-ietf-rats-endorsements.md
Comment (henkbirkholz) [inline draft-ietf-rats-endorsements.md:206]:
```suggestion
A Verifier receives Evidence from an Attesting Environment of an Attester that must be considered untrusted
```

File: draft-ietf-rats-endorsements.md
Comment (henkbirkholz) [reply draft-ietf-rats-endorsements.md:207]: Sorry, I read from the top down to this Section and am unsure which other text you refer to. Could you point me to the right place, please?

File: draft-ietf-rats-endorsements.md
Comment (henkbirkholz) [reply draft-ietf-rats-endorsements.md]: I checked. And then got caught up in other things. Continuing...

Appraisal of Evidence is always cut into two pieces:
1.) appraisal of Evidence provenance (most prominently signature checking in object security and peer authentication in secure channels) that identifies an Attesting Environment.
2.) appraisal of Evidence Claims that identifies or classifies a Target Environment AND are ultimately used to assess the trustworthiness that Target Environment.

So, the Attesting Environment is either endorsed or it was a Target Environment before and it was assessed to be trustworthy and now takes on the duty of an Attesting Environment. Following that line of though, on the one hand "Endorsed Identity" intuitively says "Attesting Environment" to me.

"Key Material for Attestation" could be anything, right? Text above talks about the "trust anchor actual state scenario" where the Actual State would be a trust anchor used to verify Evidence or Endorsement. The text here is talking about Evidence and somewhat Layered Attestation now. So I do not think it is about Endorsement authenticity, but only Evidence authenticity.

Could "Key Material for Attestation" simply be renamed to "Key Material for Evidence Provision and Appraisal"? Or should we massage the text to clearly include Evidence and Endorsement Authenticity?

File: draft-ietf-rats-endorsements.md
Comment (henkbirkholz) [reply draft-ietf-rats-endorsements.md:211]: I read the text as a whole a few times and it seems just to be unusual English phrasing for me. The new text is pretty terse, but it seems to be good enough.

In general in this Section Attester and Attesting Environment seem to be used synonymously. Would it break the meaning of the text if we would change "Attester" to "Attesting Environments of an Attester"? See exemplary suggestion above https://github.com/ietf-rats-wg/rats-endorsements/pull/18/files#r1625613299

File: draft-ietf-rats-endorsements.md
Comment (thomas-fossati) [reply draft-ietf-rats-endorsements.md]: > [...] Following that line of though, on the one hand "Endorsed Identity" intuitively says "Attesting Environment" to me.

Yes, exactly.  The section is all about endorsing the authenticity of the (bottom-most) attesting environment.
And, BTW, I am fine with it.

> "Key Material for Attestation" could be anything, right?

Yes, my main concern with the proposed new title is it's too broad.  We are talking about a pretty specific use of crypto, i.e., how the verifier uses it to assess evidence's provenance.
The relevant thing is not crypto key material per se, but how we tell the verifier what key is the one that verifies the authenticity of AE's evidence statement.

> Could "Key Material for Attestation" simply be renamed to "Key Material for Evidence Provision and Appraisal"?

I don't think that makes the point any clearer.
I like your breakdown above (provenance vs claims), and I think.a more precise title would be "Endorsing Evidence Provenance".

> Or should we massage the text to clearly include Evidence and Endorsement Authenticity?

No, please.  This section has a nice and narrow scope.  Let's not bloat it.

File: draft-ietf-rats-endorsements.md
Comment (thomas-fossati) [reply draft-ietf-rats-endorsements.md:206]: This hurts readability and doesn't seem to add much in terms of precision.

Comment (dthaler) [reply draft-ietf-rats-endorsements.md:206]: I agree with Thomas on this one, it's much less readable.

File: draft-ietf-rats-endorsements.md
Comment (dthaler) [reply draft-ietf-rats-endorsements.md]:
```suggestion
# Endorsing Evidence Provenance
```

Comment (dthaler) [reply draft-ietf-rats-endorsements.md]: Ok, changed to "Endorsing Evidence Provenance"

File: draft-ietf-rats-endorsements.md
Comment (thomas-fossati) [reply draft-ietf-rats-endorsements.md]: > Updated text. Is the new text better?

WFM, thanks!

**Review by thomas-fossati — APPROVED** [review]
🚢 it!

**Comment by dthaler** [comment]
Merging after sufficient time has passed after saying when I would merge this unless I heard objections.
I will submit a new draft version, and any further comments can be dealt with in a new PR if needed.

**Pull request merged**
